.github/workflows/create-tag.yml

```diff
@@ -0,0 +1,104 @@
+name: "Create a tag"
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Branch to be tagged"
+        required: true
+        type: string
+        default: main
+      tag:
+        description: "Tag for new version (v1.23.4)"
+        required: true
+        type: string
+      create_release:
+        description: "Create release and set as latest"
+        type: boolean
+        default: true
+      base_tag:
+        description: "Base tag to generate commit list for release notes"
+        required: false
+        type: string
+
+jobs:
+  create-tag:
```

Comment:
Please confirm sdk-go builds correctly with the commit to be tagged before tagging the release. This is IMO an essential part of this automation, because without it, we are just making it easier to push out something broken to users.

Reply:
To clarify, not `sdk-go@master` but `sdk-go@latest`. We need to make sure our latest API tag works with our latest Go SDK tag. Granted on the Go SDK side, we should update the `api-go` before each release so the inverse is tested as well. I will circulate that requirement separately.

Reply:
The check is being done in the `api` workflow, see https://github.com/temporalio/api/pull/437
I changed this workflow to `workflow_call` event so it can't be triggered manually.

Reply:
👍 Dropping my objection to merge then.